proto/sentry_protos/snuba/v1/request_common.proto

```diff
@@ -0,0 +1,32 @@
+syntax = "proto3";
+
+package sentry_protos.snuba.v1;
+
+import "google/protobuf/timestamp.proto";
+import "sentry_protos/snuba/v1/trace_item_filter.proto";
+
+message RequestMeta {
+  uint64 organization_id = 1;
+  string cogs_category = 2;
+  string referrer = 3;
+  // can be empty
+  repeated uint64 project_ids = 4;
+  google.protobuf.Timestamp start_timestamp = 5;
+  google.protobuf.Timestamp end_timestamp = 6;
+  TraceItemName trace_item_name = 7;
+}
+
+enum TraceItemName {
+  TRACE_ITEM_NAME_UNSPECIFIED = 0;
+  TRACE_ITEM_NAME_EAP_SPANS = 1;
+}
+
+message PageToken {
```

Comment:
<details>

<summary>Conversation Colin and I had about this (for historical posterity) </summary>

Colin:
does clickhouse work well with ID-based pagination? wouldn't it be hard to convert a span id to an offset to use in a subquery?
because "find span by id" is slow in current config


Volo Kluev
ah yes let's chat about that, would be good to question my assumptions here
You are right that ID-based pagination would not work
What I'm imagining is a filter that is passed back by the server that is based on the order by.
https://github.com/getsentry/sentry-protos/pull/28/files#diff-94d63309faaad438fa788b21a8031e155ff059efeec32e921996e39257226bb7R27-R30
So for example, if in the first request you ordered by (timestamp, project_name), the server would take the max timestamp and project name and pass that back in the page token


Colin Chartier
you could maybe use (timestamp, id) pairs, but that would be weird devex


Volo Kluev
the devex I'm imagining is that the server always passes back a pagetoken and all the client has to do is pass it in to paginate


Colin Chartier
could we just return like a hash string
struct pack (timestamp, span id) and then hex encode that
that's how like reddit or github pagination works

Volo Kluev
yes we could, I was thinking of that at first but that is essentially what protobuf is doing, it's binary encoding the conditions of the filter


Colin Chartier
the trace item filter one is clever, but where would it go
into a subquery?
because if your filter is (span_id = blah and timestamp = blah)
then you can't just AND that with the existing query, you need to do OFFSET (select count() from eap_spans where span_id = blah and timestamp = blah)
or something equivalent
if the condition is only on timestamp, then why not just send the timestamp instead of the whole structure

Volo Kluev
the filter would be something like (timestamp > X) and I think I could AND that with the existing query


Colin Chartier
the alternative is you can just have a list of keys, and just do like
where (org_id, project_id, _sort_timestamp, span_id) > (a, b, c, d)
which would hit the primary key
but that sort of condition can't currently be encoded by traceitemfilter
i think of these options, the one I like best is just
repeated TraceItemKey cursor = 1
or even repeated string cursor = 1


Volo Kluev
What about
(org_id > a AND project_id > b AND _sort_timestamp > c AND span_id > d)


Colin Chartier
that could work, yeah
ok, i'm fine with that and encoding it as traceitemfilter too